### DISCUSSION TITLE
Shortcuts for going in/out of subpatches

### DISCUSSION THREAD

**Discussion by franckaubry** [cables-gl/cables_docs#346] [Feature Requests] [answered]
As the awesomeness of subpatches and blueprints are being revealed to me, I find myself going in and out and in and out and in and... You get the pictures. There is a lot going in and out of subpatches.
I propose a couple of shortcuts to quickly go into a selected subpatch/blueprint and likewise quickly go out again.

Touchdesigner and Houdini uses these:
**U** → Out of subpatch / Up one level
**I** → Into subpath / Down one level

Is this an workflow enhancement other's also see the use for?

**Comment by pandrr** [comment]
ah yes, i always wanted to add a key for going up one level

but how does I work when you habe multiple child subpatches ?

**Reply by franckaubry** [reply]
Do you mean that you have selected several subpatches at the same time?
You can only select one and go into that one.

If you mean several layers of subpatches, then the logic is the same. You select a subpatch and go in/down one level with **I**. If you are deep inside several layers of subpatches already, you go out/up one level each time you press **U** 

Or did you mean something else?

**Comment by pandrr** [comment]
ah ok so i is the same as doubleclick ?
dunno if we need this, because to select an op you have to click it anyway

**Comment by franckaubry** [comment]
Ah! I kind of get your point.
The other thing Touchdesigner does is that if your middle mouse scroll far enough you "jump" out of the subpatch, and the same if you want to go in. You mouse point over the subpatch op and scroll in to "jump" into it.
(So TD gives us two ways to navigate subpatches, mouse and shortcuts)

One thing that might make more sense in a Cables context, is to have double click as a mechanism to both go in and out again.
Meaning you double-click on the op to go in.
Double-clicking on the background of a subpatch to go out again.

Maybe?

**Comment by pandrr** [comment, marked as answer]
double click is an interesting thought.
hmmm in the old patchrenderer double click zoomed out so you see the whole patchfield. need to think about this...

the U key is implemented on dev.cables.gl

**Reply by franckaubry** [reply]
I see that Illustrator also uses the double click to get into group focus mode and double click to get out of the group focus again. So the mechanism is a bit more universal than I first thought, which is a good thing. :)

**Reply by franckaubry** [reply]
I must say the "U" key is a pleasure to have ♥
